samples/algorithms/sudoku-grover/ColoringGroverWithConstraints.qs

```diff
@@ -0,0 +1,395 @@
+                                nEdges + nStartingColorConstraints .. nEdges + nStartingColorConstraints + numVertices - 1]);
+                for ((color, conflictQubit) in zippedColorAndConfictQubit) {
+                    // Only allow colors from 0 to 8 i.e. if bit #3 = 1, then bits 2..0 must be 000.
+                    using (tempQubit = Qubit()) {
+                        within {
+                            ApplyOrOracle(color[0 .. 2], tempQubit);
+                        } apply{
+                            // AND color's most significant bit with OR of least significant bits. 
+                            // This will set conflictQubit to 1 if color > 8
+                            CCNOT(color[3], tempQubit, conflictQubit);
+                        }
+                    }
+                }
+            } apply {
+                // If there are no conflicts (all qubits are in 0 state), the vertex coloring is valid
+                (ControlledOnInt(0, X))(conflictQubits, target);
+            }
+        }
+    }
+
+    /// # Summary
+    /// OR oracle for an arbitrary number of qubits in query register
+    ///
+    /// # Inputs
+    /// ## queryRegister
+    /// Qubit register to query
+    /// ## target
+    /// Target qubit for storing oracle result
+    operation ApplyOrOracle (queryRegister : Qubit[], target : Qubit) : Unit is Adj {        
+        // x₀ ∨ x₁ = ¬ (¬x₀ ∧ ¬x₁)
+        // First, flip target if both qubits are in |0⟩ state
+        (ControlledOnInt(0, X))(queryRegister, target);
+        // Then flip target again to get negation
+        X(target);
+    }
+    
+    /// # Summary
+    /// Using Grover's search to find vertex coloring.
+    ///
+    /// # Input
+    /// ## numVertices
+    /// The number of Vertices in the graph
+    /// ## bitsPerColor
+    /// The number of bits per color
+    /// ## maxIterations
+    /// An estimate of the maximum iterations needed
+    /// ## oracle
+    /// The Oracle used to find solution
+    /// 
+    /// # Output
+    /// Int Array giving the color of each vertex
+    operation FindColorsWithGrover (numVertices : Int, bitsPerColor : Int, maxIterations : Int, 
+        oracle : ((Qubit[], Qubit) => Unit is Adj)) : Int[] {
+        // This task is similar to task 2.2 from SolveSATWithGrover kata, 
+        // (see https://github.com/microsoft/QuantumKatas/tree/master/SolveSATWithGrover)
+        // but the percentage of correct solutions is potentially higher.
+        mutable coloring = new Int[numVertices];
+
+        // Note that coloring register has the number of qubits that is 
+        // twice the number of vertices (bitsPerColor qubits per vertex).
+        using ((register, output) = (Qubit[bitsPerColor * numVertices], Qubit())) {
+            mutable correct = false;
+            mutable iter = 1;
+            repeat {
+                Message($"Trying search with {iter} iterations");
+                ApplyGroversAlgorithmLoop(register, oracle, iter);
```

Comment:
It looks like this approach tries each possible number of iterations; that may be helpful to note as a comment.

Reply:
Added as comment in above suggestion